configurator/src/main.rs

```diff
@@ -72,119 +183,191 @@ pub enum Property {
+                    format!("{}", address.host().unwrap()),
+                    rest_port,
+                    macaroon_dir_path,
+                ))
+            }
+        },
+        NodeType::CLightning => match s9_node.clone().connection_settings {
+            S9NodeConnectionSettings::Internal => Ok((
+                format!("{}", "c-lightning.embassy"),
+                3001,
+                PathBuf::from("/mnt/c-lightning"),
+            )),
+            S9NodeConnectionSettings::External {
                 address,
                 rest_port,
                 macaroon,
-                cert: _,
             } => {
-                std::fs::create_dir_all("/root/lnd-external")?;
-                File::create("/root/lnd-external/admin.macaroon")?.write_all(
-                    &base64::decode_config(
-                        macaroon,
-                        base64::Config::new(base64::CharacterSet::UrlSafe, false),
-                    )?,
-                )?;
-
-                (
+                let macaroon_dir_path = PathBuf::from(format!("/root/cl-external-{}", node_index));
+                std::fs::create_dir_all(macaroon_dir_path.as_path())?;
+                let macaroon_path = macaroon_dir_path.join("access.macaroon");
+                File::create(macaroon_path)?.write_all(&base64::decode_config(
+                    macaroon,
+                    base64::Config::new(base64::CharacterSet::UrlSafe, false),
+                )?)?;
+
+                Ok((
                     format!("{}", address.host().unwrap()),
                     rest_port,
-                    "/root/lnd-external",
-                )
+                    macaroon_dir_path,
+                ))
             }
-        };
+        },
+    }
+}
+
+// annotateRtlNames :: [RTLNodes] -> HashMap<Name, RTLNode>
+fn get_rtl_node_map(nodes: Vec<RTLNode>) -> HashMap<String, RTLNode> {
+    let mut m = HashMap::new();
+    for node in nodes {
+        m.insert(node.ln_node.clone(), node.clone());
+    }
+    m
+}
+
+// toRtlDefault :: S9Node -> RTLNode
+fn to_rtl_default(
+    s9_node: &S9Node,
+    node_index: u8,
+    macaroon_path: PathBuf,
+    address: String,
+    rest_port: u16,
+) -> RTLNode {
+    RTLNode {
+        index: node_index,
+        ln_implementation: s9_node.typ.clone().into(),
+        ln_node: s9_node.name.clone(),
+        authentication: RTLNodeAuthentication { macaroon_path },
+        settings: RTLNodeSettings {
+            user_persona: RTLNodePersona::OPERATOR,
+            theme_mode: RTLNodeThemeMode::NIGHT,
+            theme_color: RTLNodeThemeColor::PURPLE,
+            fiat_conversion: false,
+            channel_backup_path: Some(format!("/root/backup/node-{}", node_index).into()),
+            ln_server_url: Some(format!("https://{}:{}", address, rest_port).into()),
+        },
+    }
+}
+
+// updateRTLNodeWithNewS9Data :: &S9Node -> RTLNode -> RTLNode
+fn to_rtl_from_previous(
+    prev_rtl_node: &RTLNode,
```

Comment:
take ownership here
```suggestion
    prev_rtl_node: RTLNode,
```

Reply:
This causees line 339 (`rtl_node_map.get(&s9_node.name).unwrap(),`) not to compile:

```
mismatched types
expected struct `RTLNode`, found `&RTLNode`rustc[E0308](https://doc.rust-lang.org/error-index.html#E0308)
```